public/app/plugins/datasource/loki/APP_PLUGIN_DOCUMENTATION.md

```diff
@@ -0,0 +1,148 @@
+# Loki data source documentation for app plugins developers
+
+Welcome to the developer documentation for the Loki data source! This documentation is designed to assist app plugin developers in leveraging the power of Loki data source within Grafana. Loki data source provides a set of APIs to simplify common tasks, making it easier to create engaging app plugins.
+
+## Introduction
+
+Loki data source provides a variety of methods, but not all of them are suitable for external use. In this documentation, we will focus on the key APIs that are highly recommended for app plugin development.
+
+It's important to note that the other APIs and methods that are not mentioned here may undergo changes or may not be suitable for external integration. Therefore, we do not recommend relying on them for your development needs.
+
+## Recommended APIs
+
+e strongly advise using these recommended APIs instead of direct proxy calls because they offer several benefits:
+
+- Caching of results: These APIs intelligently cache results based on their input arguments and the currently selected time range. This caching mechanism improves performance by reducing redundant fetch requests.
+
+- Proper authentication handling: The recommended APIs correctly handle custom authentication data source options, ensuring seamless authentication when making requests. This enhances both security and data access control.
+
+- Improved instrumentation: By opting for these recommended APIs, you gain access to advanced instrumentation features. These features provide better monitoring and analysis capabilities, offering deeper insights into your data source interactions.
+
+### Fetching Loki label names
+
+`datasource.languageProvider.fetchLabels()` can be used to fetch label names from connected Loki data source. Labels are essential for organizing and filtering log data. You can use this method to retrieve labels, providing a foundation for various data manipulation tasks within your app plugin.
```

Comment:
Would it be possible to an external developer to work with custom LanguageProvider instances? Or should they always use it from a data source instance?